puppet-staging -> puppet-archive

angrox opened the pull request:
Migrated the module from puppet-staging to puppet-archive (the one oracle_java uses too). Secondary I added the possibility to define a checksum and checksum_type to test the downloaded archive file against.

antoineco commented:
Don't forget extras.pp, it also uses staging. Actually in v0.4 I switched back from archive to staging because ~~archive didn't handle jar files, if I remember well.~~

Edit: actually that was not the reason, but I can't even remember why I did that. It had something to do with the formats Archive supported back then.
You can find my changes [here](https://github.com/antoineco/aco-tomcat/commit/5c4ab7a885e28831b68adc3ceb6edeb1ee0fb93d#diff-27faa283b1a26b458d1c8dd429985f73L9).

angrox commented:
Ok, extra.pp needs the change, too. I will test with jar files., haven't done that
Edit: I have seen you already change extra.pp. I will test the changes and the deployment of jar files (tomorrow I have time for that)

antoineco commented:
The changes are actually from long time ago, you basically have to revert them :)

angrox commented:
Let's put this on hold for a few days. Looks like the puppet-archive module moved to the puppet-community. I am currently asking the developer if the module will be republished on the forge too.

antoineco commented:
It's here https://forge.puppetlabs.com/puppet/archive

angrox commented:
No. The new Repository is here: https://github.com/puppet-community/puppet-archive
That is the problem: It is completly unclear to me if the puppet-community (https://puppet.community/) is a community driven fork from the puppet forge. Hopefully they are also pushing their modules back to the forge and change the project url. If not it will leave a lot of forge modules with unsatisfied dependencies.

antoineco commented:
Yes it's the same, it was forked and published under a different name, that's all.

antoineco commented:
Hi @angrox, looks like I didn't get any notification for your new commits, sorry! Is that ready for review now?

angrox commented:
Hi antoineco, 

sorry, I will continue my work on this. I will update my codebase and then try to bring the new patch up - please give me a day or two.

antoineco commented:
No rush, just mention my name to catch my attention if needed :)

angrox commented:
I will do. Thanks for your patience!

angrox commented:
@antoineco Am I correct that the rspec tests via travis-ci are always failing?

antoineco commented:
I think it's not even running because of conflicts with the master branch. You could try to rebase your branch and fix conflicts manually, maybe that would help.

angrox commented:
@antoineco, I am ready that you have a look at it. I have changed the following: 
- Moved from puppet/staging to puppet/archive
- Fixed spec_helper.rb so local rspec tests will succeed (manifest and module directory)
- Added puppetversion a fact in spec_helper.rb for in puppet 4.3.x the fact is not prefilled when using puppet-rspec
- Changed the test against Fedora in archive_spec.rb from F21 to F23 with tomcat8
- Merged the latest commits

antoineco commented on spec/spec_helper.rb:
```diff
@@ -1 +1,9 @@
 require 'puppetlabs_spec_helper/module_spec_helper'
+require 'rspec-puppet'
```
This is implicitly included by the spec_helper above, probably also the whole block below (not sure about that)

angrox replied:
I just tested it. You are right that the "require rspec-puppet" is obsolete (it is included in the puppetlabs spec helper). 
Nevertheless the rake task does not succeed if the directories for the fixtures are not defined (at least in my local environment).

antoineco replied:
Strange thing, it works for me with both Puppet 4.3 and 3.8 (fresh system running in Docker). I think you have some other Rspec settings interfering. Are you running the test with `bundle exec rake spec`?

angrox replied:
Yes "bundle exec rake spec". Tested in my rvm environment with ruby 1.9.3 and 2.2.4

antoineco commented on metadata.json:
```diff
@@ -21,8 +21,8 @@
     {
-      "name": "nanliu/staging",
-      "version_requirement": ">= 1.0.0 < 2.0.0"
+      "name": "puppet-community/archive",
```
Shouldn't it be `puppet/archive`?

angrox replied:
Yes, you are right. That changed from the last time I edited it. All puppet-community forge modules were renamed.

angrox commented:
Heya, I just wanted to ask how we want to proceed? I am a little bit confused right now :-|

antoineco commented:
Hey @angrox, could you reset `spec_helper.rb` and squash all the commits into one? After that I can review one more time and merge it :+1:

angrox commented:
Done, as good as I could!

antoineco commented:
Hm, I still see 10 commits, have you done a `git rebase -i HEAD~10`? From there you should replace all the `pick` by `s` or `squash`, but the first one. Example:

```
pick 916c7cb commit message
s a8627e2 commit messages
s f9ae49d commit message
s 5be5d1e commit message
[...]
s 6fe053a commit message
s 9371cd8 commit message
```

angrox commented:
Sorry for the delay. I have rebased everything you mentioned. Could you please check it again? Thanks!

antoineco commented:
Hey @angrox, I still see 14 commits unfortunately, did you push your changes?

angrox commented:
Strange. As soon as I clone the repository I see only one commit from me where everything is in.

angrox commented:
Ok. It's getting to complex. Let's close this pull request. I have a complete patch of my change, up to the current commit state of your master branch. I will open a new pull request where we have one clear commit.
